File: src/client/src/types/blocks/terminalTypeReferenceView.ts
import { HasCardinality } from "../common/hasCardinality";
import { Direction } from "../terminals/direction";
import { TerminalView } from "../terminals/terminalView";
import { ConnectionPoint } from "./connectionPoint";

export interface TerminalTypeReferenceView extends HasCardinality {
  direction: Direction;
  terminal: TerminalView;
  connectionPoint: ConnectionPoint | null;
}

File: src/server/Tyle.Api/Blocks/TerminalTypeReferenceView.cs
using Tyle.Api.Terminals;
using Tyle.Core.Blocks;
using Tyle.Core.Common;
using Tyle.Core.Terminals;

namespace Tyle.Api.Blocks;

public class TerminalTypeReferenceView : HasCardinality
{
    public Direction Direction { get; set; }
    public required TerminalView Terminal { get; set; }
    public ConnectionPoint? ConnectionPoint { get; set; }
}